### PR TITLE
fix various lint problems

### PR DESCRIPTION
**Pull request by mvladev**
**What this PR does / why we need it**:

- capitalized return error strings
- `if someBool != true { ... }`
- `strings.EqualFold` usage
- dead code

etc

**Which issue(s) this PR fixes**:
Fixes # n/a

**Special notes for your reviewer**:

**Release note**:

```improvement operator
NONE
```

### CODE REVIEW

File: pkg/utils/timewindow_test.go [hunk 1]
```diff
@@ -54,11 +49,14 @@ var _ = Describe("utils", func() {
 
 		Describe("#ParseMaintenanceTime", func() {
 			It("should return the correctly parsed maintenance time", func() {
+				const (
+					cetOffset = 2
+					hour      = 16
+					minute    = 5
+					second    = 54
+				)
 				var (
```
Comment (timuthy) [inline pkg/utils/timewindow_test.go]: Remove `var` block.

File: cmd/gardener-scheduler/main.go [hunk 2]
```diff
@@ -18,24 +18,23 @@ import (
...
-func init() {
-	features.RegisterFeatureGates()
-}
-
 func main() {
 	if len(os.Getenv("GOMAXPROCS")) == 0 {
 		runtime.GOMAXPROCS(runtime.NumCPU())
 	}
 
+	features.RegisterFeatureGates()
```
Comment (timuthy) [inline cmd/gardener-scheduler/main.go:33]: Why do you move this from `init()` to here and why only for the scheduler?

**Review by timuthy — CHANGES_REQUESTED** [review]

File: pkg/utils/timewindow_test.go (hunk 1 above)
Comment (mvladev) [reply pkg/utils/timewindow_test.go]: done

File: cmd/gardener-scheduler/main.go (hunk 2 above)
Comment (mvladev) [reply cmd/gardener-scheduler/main.go:33]: Why does it needs `init()` in the first place when it was in the `main` package?

> why only for the scheduler?

I was fixing a deprecation warning (package change). Didn't look into other components.

Comment (timuthy) [reply cmd/gardener-scheduler/main.go:33]: I'm not opposing in general, just voting for doing it in a consistent way.
Do you mind adjusting the other places as well?

Comment (mvladev) [reply cmd/gardener-scheduler/main.go:33]: Already done

**Review by timuthy — APPROVED** [review]

**Pull request merged**